wikipedia_test.go

```diff
@@ -0,0 +1,85 @@
+package meteo_test
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/qba73/meteo"
+)
+
+func TestNewWikipediaClient(t *testing.T) {
+	t.Parallel()
+	var err error
+	var c *meteo.WikipediaClient
+	c, err = meteo.NewWikipediaClient("UserName")
+	if err != nil {
+		t.Fatal(err)
+	}
+	_ = c
+}
+
+func TestNewWikipediaClientWithoutUserName(t *testing.T) {
+	t.Parallel()
+	_, err := meteo.NewWikipediaClient("")
+	if err == nil {
+		t.Fatal(err)
+	}
+}
+
+func TestNewWikipediaClientWithUserName(t *testing.T) {
+	t.Parallel()
+	got, err := meteo.NewWikipediaClient("UserName")
+	if err != nil {
+		t.Fatal(err)
+	}
+	want := "UserName"
+	if want != got.UserName {
+		t.Errorf("want %s, got %s", want, got.UserName)
+	}
+}
+
+func TestGetCoordinatesSingleGeoName(t *testing.T) {
+	t.Parallel()
+
+	testFile := "testdata/response-geoname-wikipedia.json"
+
+	ts := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		wantReqURL := "/wikipediaSearchJSON?maxRows=10&q=Castlebar&username=UserName"
+		gotReqURL := r.RequestURI
+		if wantReqURL != gotReqURL {
+			t.Errorf("want %q for wikipedia URL, got %q", wantReqURL, gotReqURL)
+		}
+		f, err := os.Open(testFile)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer f.Close()
+		io.Copy(rw, f)
+	}))
+	defer ts.Close()
+
+	c, err := meteo.NewWikipediaClient("UserName")
+	if err != nil {
+		t.Fatal(err)
+	}
+	c.BaseURL = ts.URL
+
+	got, err := c.GetCoordinates("Castlebar", "IE")
+	if err != nil {
+		t.Fatalf("GetCoordinates('Castlebar', 'IE') got err %v", err)
+	}
+	want := meteo.Place{
+		Lng:         -9.2988,
```

Comment:
It's possible we might run into a floating-point imprecision issue comparing these values, isn't it? If Wikipedia provides four decimal places of precision, what about storing the lat and long as `int`:

```go
Longitude: -92988,
Latitude: 538608
```

You can always insert the decimal point for display.